app/database/user_item_store_compute_all_user_items.go

```diff
@@ -98,60 +98,89 @@ func (s *UserItemStore) ComputeAllUserItems() (err error) {
+										 LEFT JOIN best_scores AS task_children ON(
+												items_items.child_item_id = task_children.item_id AND
+												task_children.group_id = parent_groups_attempts.group_id
+								)
+										 JOIN items ON(
+										items.ID = items_items.child_item_id
+								)
+						WHERE items.type <> 'Course' AND items.no_score = 0
+						GROUP BY parent_groups_attempts.id
+					)
+					UPDATE groups_attempts
 					LEFT JOIN (
 						SELECT
 							MAX(children.latest_activity_at) AS latest_activity_at,
 							SUM(children.tasks_tried) AS tasks_tried,
 							SUM(children.tasks_with_help) AS tasks_with_help,
 							SUM(children.tasks_solved) AS tasks_solved,
 							SUM(validated) AS children_validated,
-							children.user_id AS user_id,
+							children.group_id AS group_id,
 							items_items.parent_item_id AS item_id
-						FROM users_items AS children 
+						FROM groups_attempts AS children 
 						JOIN items_items ON items_items.child_item_id = children.item_id
-						GROUP BY children.user_id, items_items.parent_item_id
+						GROUP BY children.group_id, items_items.parent_item_id
 					) AS children_data
-						USING(user_id, item_id)
+						USING(group_id, item_id)
 					LEFT JOIN task_children_data_view AS task_children_data
-						ON task_children_data.user_item_id = users_items.id
+						ON task_children_data.id = groups_attempts.id
 					JOIN items
-						ON users_items.item_id = items.id
+						ON groups_attempts.item_id = items.id
 					LEFT JOIN items_items
-						ON items_items.parent_item_id = users_items.item_id
+						ON items_items.parent_item_id = groups_attempts.item_id
 					SET
-						users_items.latest_activity_at = IF(task_children_data.user_item_id IS NOT NULL AND
-							children_data.user_id IS NOT NULL AND items_items.id IS NOT NULL,
-							children_data.latest_activity_at, users_items.latest_activity_at),
-						users_items.tasks_tried = IF(task_children_data.user_item_id IS NOT NULL AND
-							children_data.user_id IS NOT NULL AND items_items.id IS NOT NULL,
-							children_data.tasks_tried, users_items.tasks_tried),
-						users_items.tasks_with_help = IF(task_children_data.user_item_id IS NOT NULL AND
-							children_data.user_id IS NOT NULL AND items_items.id IS NOT NULL,
-							children_data.tasks_with_help, users_items.tasks_with_help),
-						users_items.tasks_solved = IF(task_children_data.user_item_id IS NOT NULL AND
-							children_data.user_id IS NOT NULL AND items_items.id IS NOT NULL,
-							children_data.tasks_solved, users_items.tasks_solved),
-						users_items.children_validated = IF(task_children_data.user_item_id IS NOT NULL AND
-							children_data.user_id IS NOT NULL AND items_items.id IS NOT NULL,
-							children_data.children_validated, users_items.children_validated),
-						users_items.validated = IF(task_children_data.user_item_id IS NOT NULL AND items_items.id IS NOT NULL,
+						groups_attempts.latest_activity_at = IF(task_children_data.id IS NOT NULL AND
```

Comment:
It is a hack for setting groups_attempts.ancestors_computation_state = 'done' in the same query (and keeping all other columns unchanged when there where no children found for the item). In PHP they had a separate query for setting groups_attempts.ancestors_computation_state.

Reply:
Ok, I get why it is equivalent to what was done on the users_items in PHP. But I still don't get the point of not changing the former value when no children. For instance, if its last child has been removed, I don't see why it should store the former "validated" from his child for instance. I think it is more something forgotten before than a "feature". Do you see any reasons?

And in any case, I do not see the benefit of rejoining items_items

Reply:
Leaves never have children and I doubt we want to wipe leaves' data with zeros.

Reply:
For `And in any case, I do not see the benefit of rejoining items_items`, it's fixed now.

Reply:
Here the propagation goes upwards, so it is not about reaching leaves but chapter etc with only "no_score" children. 
I asked the question twice, you haven't found reasons from what I understood. 
So, if we do not find reason (I don't say there isn't), I think we may put more appropriate values when there are no children. In addition to fixing a potential problem, it may make the request more understandable. So I would say:
```sql 
target_groups_attempts.latest_activity_at =
  GREATEST(
       IFNULL(children_stats.latest_activity_at, target_groups_attempts.latest_activity_at),
       IFNULL(target_groups_attempts.latest_activity_at, children_stats.latest_activity_at)
   ), -- GREATEST which returns the other value if one is null
target_groups_attempts.tasks_tried = IFNULL(children_stats.tasks_tried, 0),
target_groups_attempts.tasks_with_help = IFNULL(children_stats.tasks_with_help, 0),
target_groups_attempts.tasks_solved = IFNULL(children_stats.tasks_solved, 0),
target_groups_attempts.children_validated = IFNULL(children_stats.children_validated, 0),
target_groups_attempts.validated = CASE ... END,
target_groups_attempts.validated_at = 
  IF(items.validation_type = 'Categories', children_stats.max_validated_at_categories, children_stats.max_validated_at),
target_groups_attempts.ancestors_computation_state = 'done'
```
... which is what makes more sense to me for propagation across attempts. (slightly different from what was in php)

Reply:
> Here the propagation goes upwards, so it is not about reaching leaves but chapter etc with only "no_score" children.

Reply:
The proposed query would wipe `task_tried`, `tasks_with_help`, `tasks_solved`, `validated`, `validated_at` of leaves with zeros/NULLs.